cmis_write/ir_attachment.py

```diff
@@ -0,0 +1,283 @@
+        if context is None:
+            context = {}
+        location = self.pool['ir.config_parameter'].get_param(
+            cr, uid, 'ir_attachment.location')
+        file_size = len(value.decode('base64'))
+        if location:
+            attach = self.browse(cr, uid, id, context=context)
+            if attach.store_fname:
+                self._file_delete(cr, uid, location, attach.store_fname)
+            fname = self._file_write(cr, uid, location, value)
+            # SUPERUSER_ID as probably don't have write access,
+            # trigger during create
+            super(ir_attachment, self).write(
+                cr, SUPERUSER_ID, [id],
+                {'store_fname': fname, 'file_size': file_size},
+                context=context)
+        else:
+            # Don't save datas, put db_datas field to none
+            super(ir_attachment, self).write(
+                cr, SUPERUSER_ID, [id],
+                {'db_datas': None, 'file_size': file_size}, context=context)
+        return True
+
+    def _data_get(self, cr, uid, ids, name, arg, context=None):
+        """
+        Get data from DMS
+        """
+        if context is None:
+            context = {}
+        result = {}
+        location = self.pool['ir.config_parameter'].get_param(
+            cr, uid, 'ir_attachment.location')
+        bin_size = context.get('bin_size')
+        for attach in self.browse(cr, uid, ids, context=context):
+            if location and attach.store_fname:
+                result[attach.id] = self._file_read(
+                    cr, uid, location, attach.store_fname, bin_size)
+            elif attach.id_dms:
+                datas = self.action_download(
+                    cr, uid, attach.id, context=context)
+                if datas:
+                    result[attach.id] = datas
+                else:
+                    result[attach.id] = ''
+                    _logger.warn('Access error of DMS')
+            else:
+                result[attach.id] = ''
+                _logger.warn('Attachment %s has no id_dms', attach.name)
+        return result
+
+    _columns = {
+        'id_dms': fields.char('Id of Dms', help="Id of Dms."),
+        'download_id': fields.one2many(
+            'ir.attachment.download',
+            'attachment_id',
+            'Attachment download',
+            help="Attachment download"
+        ),
+        'datas': fields.function(
+            _data_get,
+            fnct_inv=_data_set,
+            string='File Content',
+            type="binary",
+            nodrop=True
+        ),
+        'attachment_document_ids': fields.one2many(
+            'ir.attachment.doc.backend',
+            'attachment_id',
+            'Records',
+            help="Records"
+        ),
+    }
+
+
+@job
+def create_doc_in_edm(session, model_name, value, res,
+                      dict_metadata, user_login, filters=None):
+    """
+    This method allows to create a doc from Odoo to DMS
+    """
```

Comment:
@ehdem You don't seem to use any connector unit. Why? Connector units are the core of the connector framework. Without using these, your code will be hard to maintain and inherit. Also, you don't make any use of the framework's exceptions. How will the worker know whether to retry a job or to cancel it?